File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(str) {
    let arr = [];
    
    let s = 0;
    let e = 0;
    let n = str.length-1;
    let max = 0;
    while(e<=n){
        
        if(!arr.includes(str[e])){
           arr.push(str.charAt(e));
           max = Math.max(arr.length,max);
           e++;
        }else{
            // console.log(arr.length);
               arr.shift();
            // console.log(arr.length);
               s++;
        }   
    }
    // console.log(arr);
        return max;
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String str) {
        List<Character> ans = new ArrayList<>();
        int s= 0;
        int e = 0;
        int n = str.length();
        int max =0;
        while(e<=n-1){
            if(!ans.contains(str.charAt(e))){
                ans.add(str.charAt(e));
                max = Math.max(max,ans.size());
                e++;
            }else{
               ans.remove(Character.valueOf(str.charAt(s)));                    
                s++;
            }
            
        }
        return max ;
    }
}